### PR TITLE
Use the original Maven Scm object to determine the browsable URL

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 1]
```diff
@@ -218,13 +218,14 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
...
         fun parseVcsInfo(mavenProject: MavenProject): VcsInfo {
             // When asking Maven for the SCM URL of a POM that does not itself define an SCM URL, Maven returns the SCM
-            // URL of the parent POM (if any) and appends the child POM's artifactId to it. This behavior is
-            // fundamentally broken because it invalidates the URL for many SCMs that cannot clone / checkout a specific
-            // path from a repository. Also, the assumption that the source code for a child artifact is stored in a
-            // top-level directory named like the artifactId inside the parent artifact's repository is often not
-            // correct.
-            // To fix this, determine the SCM URL of the root parent (if there are parents) and use that as the child's
-            // SCM URL.
+            // URL of the first parent POM (if any) that deines one and appends the artifactIds of all child POMs to it,
```
Comment (fviernau) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt]: As there is just one parent, how about using the term `ancestor` instead? (same again in line 227) e.g.: "Maven return the SCM URL of the closest ancestor that declares one.." 

type: deines ->  defines / declares

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt]: I think it's better to stay with "parent" because that's the term Maven uses.

Comment (fviernau) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt]: Maybe [this](https://maven.apache.org/plugins-archives/maven-dependency-plugin-2.9/display-ancestors-mojo.html) counts as Maven terminology as well?

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 2]
```diff
@@ -141,7 +141,35 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
...
+        fun getOriginalScm(mavenProject: MavenProject): Scm? {
+            // When asking Maven for the SCM URL of a POM that does not itself define an SCM URL, Maven returns the SCM
+            // URL of the first parent POM (if any) that deines one and appends the artifactIds of all child POMs to it,
+            // separated by slashes.
+            // This behavior is fundamentally broken because it invalidates the SCM URL for all VCS that cannot limit
+            // cloning to a specific path within a repository, or use a different syntax for that. Also, the assumption
+            // that the source code for a child artifact is stored in a top-level directory named like the artifactId
+            // inside the parent artifact's repository is often not correct.
+            // To address this, determine the SCM URL of the parent (if any) that is closest to the root POM and whose
+            // SCM URL still is a prefix the child POM's SCM URL.
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:153]: Can you move this comment to the KDoc of the function as part of this commit?

File: analyzer/src/main/kotlin/managers/Maven.kt [hunk 3]
```diff
@@ -152,7 +152,9 @@ class Maven(
             workingDir
         }
 
-        val vcsFallbackUrls = listOfNotNull(mavenProject.scm?.url, mavenProject.url)
+        val browsableScmUrl = MavenSupport.getOriginalScm(mavenProject)?.url
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Maven.kt:155]: Commit message: This is actually a fixup for 051a60b.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Maven.kt:155]: Done.

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt]: I also prefer to stick with "parent" as that's [the name of the XML tag](http://maven.apache.org/pom.html#Inheritance).

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:153]: Done.

**Comment by sschuberth** [comment]
Merging despite the unrelated AppVeyor test failure in PhpComposerTest.

**Pull request merged**